Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -551,13 +551,13 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
   createDirectoryIfMissingVerbose verbosity True libTargetDir
   -- TODO: do we need to put hs-boot files into place for mutually recursive
   -- modules?
-  let cLikeFiles  = fromNubListR $ mconcat
+  let cLikeSources  = fromNubListR $ mconcat
                       [ toNubListR (cSources   libBi)
                       , toNubListR (cxxSources libBi)
                       , toNubListR (cmmSources libBi)
                       , toNubListR (asmSources libBi)
                       ]
-      cObjs       = map (`replaceExtension` objExtension) cLikeFiles
+      cObjs       = map (`replaceExtension` objExtension) cLikeSources
```

Comment:
this is still `cObjs`, which is confusing as it's `cLikeObjs`. Ditto `cProfObjc` etc (which include objects **not only** from C-code, but from **C-like** code, where likeness is vague notion as well).

TL;DR this is not what I suggested.

Reply:
I tend to agree here. Because we don't document, we have to improve the state of variable names. Though, preferably, I'd like both a docstring and better names.